scripts/obtain/install.ps1

```diff
@@ -3,106 +3,265 @@
 # Licensed under the MIT license. See LICENSE file in the project root for full license information.
 #
 
+<#
+.SYNOPSIS
+    Installs dotnet cli
+.DESCRIPTION
+    Installs dotnet cli. If dotnet installation already exists in the given directory
+    it will update it only if the requested version differs from the one already installed.
+.PARAMETER Channel
+    Default: preview
+    Channel is the way of reasoning about stability and quality of dotnet. This parameter takes one of the values:
+    - future - Possibly unstable, frequently changing, may contain new finished and unfinished features
+    - preview - Pre-release stable with known issues and feature gaps
+    - production - Most stable releases
+.PARAMETER Version
+    Default: latest
+    Represents a build version on specific channel. Possible values:
+    - 4-part version in a format A.B.C.D - represents specific version of build
+    - latest - most latest build on specific channel
+    - lkg - last known good version on specific channel
+    Note: LKG work is in progress. Once the work is finished, this will become new default
+.PARAMETER InstallDir
+    Default: %LocalAppData%\Microsoft\.dotnet
+    Path to where to install dotnet. Note that binaries will be placed directly in a given directory.
+.PARAMETER Architecture
+    Default: auto - this value represents currently running OS architecture
+    Architecture of dotnet binaries to be installed.
+    Possible values are: auto, x64 and x86
+.PARAMETER DebugSymbols
+    If set the installer will include symbols in the installation.
+.PARAMETER DryRun
+    If set it will not perform installation but instead display what command line to use to consistently install
+    currently requested version of dotnet cli. In example if you specify version 'latest' it will display a link
+    with specific version so that this command can be used deterministicly in a build script.
+    It also displays binaries location if you prefer to install or download it yourself.
+.PARAMETER NoPath
+    By default this script will set environment variable PATH for the current process to the binaries folder inside installation folder.
+    If set it will display binaries location but not set any environment variable.
+.PARAMETER Verbose
+    Displays diagnostics information.
+.PARAMETER AzureFeed
+    Default: https://dotnetcli.blob.core.windows.net/dotnet
+    This parameter should not be usually changed by user. It allows to change URL for the Azure feed used by this installer.
+#>
+[cmdletbinding()]
 param(
-   [string]$Channel="dev",
-   [string]$version="Latest",
-   [string]$Architecture="x64"
+   [string]$Channel="preview",
+   [string]$Version="Latest",
+   [string]$InstallDir="<usershare>",
```

Comment:
Is there a reason we have to use a special sentinel like this? Why not just use null?